### ISSUE TITLE
Nonphysical overshoots in concentrations with solute transport in v1.5.1

### ISSUE THREAD

**Issue by dasvyat** [amanzi/ats#286]
as the water table drops in the uphill portion of the domain from the initial position due to drainage in the bottom, the concentrations of the tracer increase above 1. That cannot be. At the same time concentrations of the tracer in the downhill portion of the domain decrease below zero, as that area's liquid saturation increases so that the initial water can exit through the surface . This is apparent as early as time = 3 days (concentration.png , note: this was obtained in a serial run to avoid mixing in the parallel issues) but gets worse as time advances.
![395931866-e034e1c6-59f7-4fc7-86f4-b11b5de19167](https://github.com/user-attachments/assets/9832acec-6e22-490a-bdb0-aaf6dd3a00f3)

**Comment by smolins** [comment]
Agree. As I said in the original issue, I was not sure whether they were related.  What is clear is that this issue (overshoot) is more important and obscures the parallel issue. Again, I am not clear whether the parallel issue has to do with parallel runs or is it that parallel runs make it more evident.

**Comment by smolins** [comment]
Hi all

I would like to follow up on this issue.

I wonder what kind of interpolation is proposed for this, e.g. is it linear
or other options are explored?

Also, what branch is this being worked on?

Thanks

Sergi

On Fri, Dec 20, 2024 at 7:57 AM Coon, Ethan ***@***.***> wrote:

> I have a branch already that fixes the interpolation problem I think.  I
> need to get back on that branch and put together a PR.
>
>
>
> Ethan
>
>
>
> *From: *Le, Phong ***@***.***>
> *Date: *Thursday, December 19, 2024 at 1:10 PM
> *To: *Svyatsky, Daniil ***@***.***>, Coon, Ethan ***@***.***>
> *Cc: *Sergi Molins Rafa ***@***.***>, Moulton, John David <
> ***@***.***>
> *Subject: *Re: Subcycling trasnport
>
> Yes, it is the same approach in the current master branch. Let’s create a
> transport-fix branch from the current master for this issue.
>
>
>
> --
>
> *Phong V.V. Le, Ph.D.*
>
> Research Scientist
>
> Environmental Sciences Division
>
> Oak Ridge National Laboratory
>
> Email: ***@***.***
>
>
>
>
>
> *From: *Svyatsky, Daniil ***@***.***>
> *Date: *Thursday, December 19, 2024 at 2:59 PM
> *To: *Le, Phong ***@***.***>, Coon, Ethan ***@***.***>
> *Cc: *Sergi Molins Rafa ***@***.***>, Moulton, John David <
> ***@***.***>
> *Subject: *[EXTERNAL] Re: Subcycling trasnport
>
> Hi Phong,
>
>
>
> Is it the same approach in the current master or is it different in the
> current master?
>
>
>
> Thanks,
>
> Daniil.
> ------------------------------
>
> *From:* Svyatsky, Daniil ***@***.***>
> *Sent:* Thursday, December 19, 2024 12:12 PM
> *To:* Le, Phong ***@***.***>; Coon, Ethan ***@***.***>
> *Cc:* Sergi Molins Rafa ***@***.***>; Moulton, J. David <
> ***@***.***>
> *Subject:* Re: Subcycling trasnport
>
>
>
> Thank you  Phong, for confirming this.
>
> In my opinion, currently this is the reason for overshoots in issue
> (#285/#286).
>
>
>
> Daniil.
> ------------------------------
>
> *From:* Le, Phong ***@***.***>
> *Sent:* Thursday, December 19, 2024 11:56 AM
> *To:* Svyatsky, Daniil ***@***.***>; Coon, Ethan ***@***.***>
> *Subject:* [EXTERNAL] Re: Subcycling trasnport
>
>
>
> Hi Danill,
>
>
>
> You are right. For mass conservation, saturation is not interpolated in
> the case of subcycling and it is taken at Tags::NEXT and Tags::CURRENT.
>
> Ethan can explain this issue in more details.
>
>
>
> Thanks
>
> Phong
>
>
>
> --
>
> *Phong V.V. Le, Ph.D.*
>
> Research Scientist
>
> Environmental Sciences Division
>
> Oak Ridge National Laboratory
>
> Email: ***@***.***
>
>
>
>
>
> *From: *Svyatsky, Daniil ***@***.***>
> *Date: *Thursday, December 19, 2024 at 1:46 PM
> *To: *Coon, Ethan ***@***.***>, Le, Phong ***@***.***>
> *Subject: *[EXTERNAL] Subcycling trasnport
>
> Hi Ethan and Phong,
>
>
>
> I have a question. In the case of subcycling is there a place in the code
> (1.5.1) where saturation is interpolated for subcycling_current and
> subcycling_next tags? As I see it now, saturation is not interpolated and
> is taken at  Tags::NEXT and Tags::CURRENT. Am I wrong or is it guided via
> input file?
>
>
>
> Thanks,
>
> Daniil.
>
